### DISCUSSION TITLE
How to create a slideshow

### DISCUSSION THREAD

**Discussion by DotCat1985** [OrchardCMS/OrchardCore#8655] [Q&A]
How can I create a slideshow with more items like this?

![slideshow](https://user-images.githubusercontent.com/61869142/108772815-c906b380-755d-11eb-9fc6-b11c70d56f81.png)

I would like to use content types, content parts and/or something else.

**Comment by ns8482e** [comment]
You can create slide show with any JavaScript library of your choice with Orchard Content Management. 

In orchard core, 

1. Create source for carousel
- It could be  either a Content Part with fields, a Content Type with fields, a BagPart in a Content Type,  a List or it can be a Media field. 
- You can also define and use Taxonomies, Indexes, Query, Graphql to filter your source of carousel

2. Create template for the content item or part or fields that you are using for carousel in Liquid or Razor.  [See docs, how to access queries in liquid](https://docs.orchardcore.net/en/dev/docs/reference/modules/Queries/#liquid-templates) 

Here's quick and dirty example of using bootstrap carousel using agency theme for client's section (that uses BagPart). 


![carousel](https://user-images.githubusercontent.com/23270244/109010570-8c2de000-7675-11eb-9a5f-9ccb27d8385d.gif)

Here, code is updated for client section in `Content__LandingPage`  template with following code for bootstrap carousel.

```
{% if Model.ContentItem.Content.Clients.ContentItems.size > 0 %}
<!-- Clients -->
<section class="page-section py-5">
    <div id="carouselExampleSlidesOnly"  class="carousel slide" data-ride="carousel">
        <div class="carousel-inner">
            {% for client in Model.ContentItem.Content.Clients.ContentItems %}
            <div class="carousel-item">
                <a href="{{ client.Client.Url.Text }}">
                    <img class="img-fluid d-block mx-auto" src="{{ client.Client.Logo.Paths.first | asset_url }}" alt="">
                </a>
            </div>
            {% endfor %}
        </div>
    </div>
</section>
{% endif %}
```
